### PR TITLE
Add `--logging-format` flag for kube-apiserver

### PR DESCRIPTION
**Pull request by tahsinrahman**
/kind feature

**What this PR does / why we need it**:
Ref: https://github.com/kubernetes/enhancements/issues/1602

```release-note
NONE
```

/assign @serathius

### CODE REVIEW

File: cmd/kube-apiserver/app/options/options.go [hunk 1]
```diff
@@ -149,6 +152,7 @@ func (s *ServerRunOptions) Flags() (fss cliflag.NamedFlagSets) {
 	s.EgressSelector.AddFlags(fss.FlagSet("egress selector"))
 	s.Admission.AddFlags(fss.FlagSet("admission"))
 	s.Metrics.AddFlags(fss.FlagSet("metrics"))
+	s.Logs.AddFlags(fss.FlagSet("logging-format"))
```
Comment (serathius) [inline cmd/kube-apiserver/app/options/options.go:155]:
```suggestion
	s.Logs.AddFlags(fss.FlagSet("logs"))
```
We register a flag group for logs, maybe list of flags will grow in future.

**Comment by serathius** [comment]
One nit, else looks good!
Thanks for contributing

File: cmd/kube-apiserver/app/options/options.go (hunk 1 above)
Comment (tahsinrahman) [reply cmd/kube-apiserver/app/options/options.go]: thanks! updated!

**Comment by lavalamp** [comment]
/ok-to-test

Is this where the flag is defined? https://github.com/kubernetes/component-base/blob/master/logs/options.go#L53

If so, that needs some work before we can add it to apiserver. Specifically, the help message needs to explain what the flag does, and what the options are.

Also is "text" the only option? I would prefer to not install the flag until there's additional options.

**Comment by serathius** [comment]
@tahsinrahman please run ./hack/update-bazel.sh
 
@lavalamp Although this flag currently allows only "text" option, it allows for registering additional formats by registering own implementation, making logging in k8s much more customizable. I don't think we should be blocking on implementing additional formats as they will largely delay KEP implementation.

**Comment by lavalamp** [comment]
Who can register additional formats? End users? Or we, the k8s authors? If the latter I think it makes more sense to install the flag when we add a second option. I don't think that will change the time at which users get value for this, and it won't leave a confusing intermediate state.

**Comment by serathius** [comment]
Json format was added. https://github.com/kubernetes/kubernetes/issues/91490
PR to update help listing available options is in flight. https://github.com/kubernetes/kubernetes/pull/92177

I think we can proceed with this PR. @lavalamp PTAL

**Comment by lavalamp** [comment]
Great-- I'd like the help PR to merge first as well.

**Comment by serathius** [comment]
PR updating flag was was merged. Can go move forward with this one? @lavalamp

**Comment by serathius** [comment]
Looks like US people are on holiday. @wojtek-t can you take a look?

**Comment by serathius** [comment]
Hey @sttts Can you take a look? Other approves are on holidays.

**Pull request merged**